core/src/main/java/com/linecorp/armeria/client/endpoint/EndpointSelectionStrategy.java

```diff
@@ -26,13 +26,37 @@
 
     /**
      * Simple round-robin strategy.
+     *
+     * @deprecated Use {@link #unweightedRoundRobin()}.
      */
-    EndpointSelectionStrategy ROUND_ROBIN = new RoundRobinStrategy();
+    @Deprecated
+    EndpointSelectionStrategy ROUND_ROBIN = RoundRobinStrategy.INSTANCE;
 
     /**
      * Weighted round-robin strategy.
+     *
+     * @deprecated Use {@link #weightedRoundRobin()}.
      */
-    EndpointSelectionStrategy WEIGHTED_ROUND_ROBIN = new WeightedRoundRobinStrategy();
+    @Deprecated
+    EndpointSelectionStrategy WEIGHTED_ROUND_ROBIN = WeightedRoundRobinStrategy.INSTANCE;
+
+    /**
+     * Returns a weighted round-robin strategy.
+     *
+     * @see #unweightedRoundRobin()
+     */
+    static EndpointSelectionStrategy weightedRoundRobin() {
+        return WeightedRoundRobinStrategy.INSTANCE;
+    }
+
+    /**
+     * Returns an unweighted round-robin strategy, which ignores {@link Endpoint#weight()}.
+     *
+     * @see #weightedRoundRobin()
+     */
+    static EndpointSelectionStrategy unweightedRoundRobin() {
```

Comment:
I think it's fine to be `roundRobin` and `weightedRoundRobin`, I don't know if I commonly hear the word unweighted round robin